internal/pkg/sylog/sylog.go

```diff
@@ -205,3 +207,39 @@ func Writer() io.Writer {
 
 	return os.Stderr
 }
+
+// ProgressCallback is a function that provides progress information copying from a Reader to a Writer
+type ProgressCallback func(int64, io.Reader, io.Writer) error
+
+// ProgressBarCallback returns a progress bar callback unless e.g. --quiet or lower loglevel is set
+func ProgressBarCallback() ProgressCallback {
```

Comment:
`ProgressBarCallback` could be in `internal/pkg/client` instead and use sylog.GetLevel() to replace :

```
if loggerLevel <= -1 {
    return nil
}
```

Reply:
Okay - I thought about that first, but put it in sylog in case it will be used somewhere else. Having checked I don't think it will - so I'll move it and revert the sylog changes which will make the diff smaller for sure.